extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+                  break;
+                case 'border-radius':
+                  setStyle(me.element, 'border-radius', styleValue);
+                  break;
+                case 'box-shadow':
+                  setStyle(me.element, 'box-shadow', styleValue);
+                  break;
+              }
+            }
+          });
+        }
+        if (media && media['aspect-ratio']) {
+          let height;
+          const box = me.element.getLayoutBox();
+          if (media['padding-bottom']) {
+            /** Apply height for media with updated "aspect-ratio" and "padding-bottom". */
+            height =
+              box.width / media['aspect-ratio'] + media['padding-bottom'];
+            height = this.addBorderHeight_(me, height);
+            me.attemptChangeHeight(height).catch(() => {});
+          } else {
+            height = box.width / media['aspect-ratio'];
+            if (Math.abs(box.height - height) > 1) {
+              /** Apply new height for updated "aspect-ratio". */
+              height = this.addBorderHeight_(me, height);
+              me.attemptChangeHeight(height).catch(() => {});
+            }
+          }
+        }
+      }
+      if (data.method === 'cancelWidget') {
+        me.attemptCollapse().catch(() => {});
+      }
+    }
+  }
+
+  /**
+   * Constructing placeholder image SRC
+   * @return {string} url of the placeholder
+   * @private
+   * */
+  constructPlaceholderSrc_() {
+    let src = null;
+    if (this.widgetId_) {
+      const url = this.base_ + this.widgetId_ + '/thumbnail';
+      src = addParamToUrl(url, 'amp', '1');
+    } else {
+      src = addParamsToUrl(this.base_ + 'api/thumbnail', {
+        'url': this.url_,
+        'key': this.key_,
+        'amp': '1',
+      });
+    }
+    return src;
+  }
+
+  /**
+   * Test component call for required params
+   * @return {null}
+   * @private
+   * */
+  parseAttributes_() {
+    userAssert(
+      this.element.getAttribute('data-id') ||
+        this.element.getAttribute('data-url'),
+      'Iframely requires either "data-id" or a pair of "data-url" and "data-key" parameters for <%s> %s',
+      TAG,
+      this.element
+    );
+    if (!this.widgetId_) {
+      if (this.url_) {
+        userAssert(
+          this.key_,
+          'Iframely data-key must also be set when you specify data-url parameter at <%s> %s',
+          TAG,
+          this.element
+        );
+      }
+      if (this.key_) {
+        userAssert(
+          this.url_,
+          'Iframely data-url must also be set when you specify data-key parameter at <%s> %s',
+          TAG,
+          this.element
+        );
+        userAssert(
+          16 < this.key_.length || this.key_.length > 256,
+          'Iframely data-key should be between 16 and 256 characters parameter at <%s> %s',
+          TAG,
+          this.element
+        );
```

Comment:
What happens without this check? I'd say throw it inside the iframe, it's a bit superfluous here.